src/main/java/net/mcreator/element/types/Overlay.java

```diff
@@ -41,6 +42,8 @@
 
 	public Procedure displayCondition;
 
+	public GridSettings gridSettings;
+
 	public Overlay(ModElement element) {
 		super(element);

```

Comment:
you need to initialize grid in constructor similar to how eg. biome does it. see https://github.com/MCreator/MCreator/blob/e9a5e8f48507cf51bf415cae2813aceb0bce2d2c/src/main/java/net/mcreator/element/types/Biome.java#L106 and https://github.com/MCreator/MCreator/blob/e9a5e8f48507cf51bf415cae2813aceb0bce2d2c/src/main/java/net/mcreator/element/types/Biome.java#L126

Then NPE will be gone as gridSettings will not be null